staging/src/k8s.io/kubectl/pkg/drain/drain_test.go

```diff
@@ -0,0 +1,416 @@
+	return &m.discovery
+}
+
+func (m *mockKubernetes) CoreV1() corev1client.CoreV1Interface {
+	return &m.core
+}
+
+func (m *mockKubernetes) PolicyV1beta1() policyv1beta1client.PolicyV1beta1Interface {
+	return &m.policy
+}
+
+type mockCore struct {
+	corev1client.CoreV1Interface
+
+	k8s  *mockKubernetes
+	pods []corev1.Pod
+}
+
+func (m *mockCore) Pods(ns string) corev1client.PodInterface {
+	return &mockCorePods{
+		k8s:       m.k8s,
+		core:      m,
+		namespace: ns,
+	}
+}
+
+type mockCorePods struct {
+	corev1client.PodInterface
+
+	k8s       *mockKubernetes
+	core      *mockCore
+	namespace string
+}
+
+func (m *mockCorePods) Get(name string, options metav1.GetOptions) (*corev1.Pod, error) {
+	pods := m.core.pods
+	for _, pod := range pods {
+		if pod.Namespace != m.namespace {
+			continue
+		}
+		if pod.Name != name {
+			continue
+		}
+
+		return pod.DeepCopy(), nil
+	}
+
+	qr := schema.GroupResource{
+		Group:    "",
+		Resource: "pods",
+	}
+
+	err := apierrors.NewNotFound(qr, name)
+	return nil, err
+}
+
+func (m *mockCorePods) Delete(name string, options *metav1.DeleteOptions) error {
+	found := false
+
+	var keepPods []corev1.Pod
+	for _, pod := range m.core.pods {
+		if pod.Namespace == m.namespace && pod.Name == name {
+			found = true
+			continue
+		}
+		keepPods = append(keepPods, pod)
+	}
+
+	if !found {
+		return fmt.Errorf("pod not found")
+	}
+
+	m.core.pods = keepPods
+	return nil
+}
+
+type mockPolicy struct {
+	policyv1beta1client.PolicyV1beta1Interface
+
+	k8s *mockKubernetes
+
+	recordEvictions []policyv1beta1.Eviction
+}
+
+func (m *mockPolicy) Evictions(ns string) policyv1beta1client.EvictionInterface {
+	return &mockPolicyEvictions{
+		k8s:       m.k8s,
+		policy:    m,
+		namespace: ns,
+	}
+}
+
+type mockPolicyEvictions struct {
+	policyv1beta1client.EvictionInterface
+
+	k8s       *mockKubernetes
+	policy    *mockPolicy
+	namespace string
+}
+
+func (m *mockPolicyEvictions) Evict(eviction *policyv1beta1.Eviction) error {
+	m.policy.recordEvictions = append(m.policy.recordEvictions, *eviction)
+
+	return m.k8s.core.Pods(eviction.Namespace).Delete(eviction.Name, &metav1.DeleteOptions{})
+}
+
+type mockDiscoveryInterface struct {
+	discovery.DiscoveryInterface
+
+	serverGroups map[string]metav1.APIGroup
+	resources    []metav1.APIResource
+}
+
+func (m *mockDiscoveryInterface) ServerGroups() (*metav1.APIGroupList, error) {
+	gl := &metav1.APIGroupList{}
+	for _, g := range m.serverGroups {
+		gl.Groups = append(gl.Groups, g)
+	}
+	return gl, nil
+}
+
+func (m *mockDiscoveryInterface) ServerResourcesForGroupVersion(groupVersion string) (*metav1.APIResourceList, error) {
+	rl := &metav1.APIResourceList{}
+	for _, r := range m.resources {
+		if r.Group != "" && r.Group+"/"+r.Version != groupVersion {
+			continue
+		}
+		if r.Group == "" && r.Version != groupVersion {
+			continue
+		}
+		rl.APIResources = append(rl.APIResources, r)
+	}
+	return rl, nil
+}
+
+func NewMockKubernetes() *mockKubernetes {
```

Comment:
> staging/src/k8s.io/kubectl/pkg/drain/drain_test.go:301:26: exported func NewMockKubernetes returns unexported type *drain.mockKubernetes, which can be annoying to use

Could just not export the mock?

Reply:
(this is why verify is failing @justinsb)

Reply:
Thanks @BenTheElder!  Although I just switched this to use the fake client-go instead